### DISCUSSION TITLE
[@types/pouchdb-core] PouchDB `Plugin` typing does not support userland plugin API

### DISCUSSION THREAD

**Discussion by jtcooper10** [DefinitelyTyped/DefinitelyTyped#57335] [Issues with a @types package] [answered]
In the [API documentation for PouchDB](https://pouchdb.com/api.html#plugins), the `PouchDB.plugin()` method accepts either a function or an object. The types definition [appears to be assigned to a string literal](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/b8fe3491fc54d7122c747b4acbf9967b436c3220/types/pouchdb-core/index.d.ts#L548), which caused a compiler error when used as-is. This was mentioned in Issue #34521.

It's entirely possible I'm missing something, please let me know!

[a comment by an automated account is not shown]

**Comment by fredgalvao** [comment, marked as answer]
I tried tracing PouchDB type definitions back to 2016 when I was involved in rewriting all of it, and I couldn't find [back then any mention of typings for the `.plugin()` side of things], nor could I find [right now any reason why the argument to `.plugin()` should be a literal]. That last part sounds like a **TODO** in code, as something that was indeed supposed to be implemented later (aka now). I'd say it should be pretty easy for you, @jtcooper10 or @assemblethis, to contribute.

**Reply by jtcooper10** [reply]
Addressed in PR #57978, if you'd like to take a look.
